### PR TITLE
Message Refactor

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Cleaning up the messages prior to implementing any further for them.

### CODE REVIEW

File: src/OpenLoco/Message.h [hunk 1]
```diff
@@ -36,10 +36,40 @@ namespace OpenLoco
         advice,
     };
 
+    enum class NewsItemSubTypes : uint8_t
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Message.h]: Probably best to stick to singular for types?

File: src/OpenLoco/Message.h [hunk 2]
```diff
@@ -58,8 +58,8 @@ namespace OpenLoco
     {
         constexpr uint16_t unk0 = 1 << 0;
         constexpr uint16_t unk1 = 1 << 1;
-        constexpr uint16_t unk2 = 1 << 2;
-        constexpr uint16_t unk3 = 1 << 3;
+        constexpr uint16_t hasFirstItem = 1 << 2;
+        constexpr uint16_t hasSecondItem = 1 << 3;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Message.h:62]: Would the use of `Entity` be appropriate here?

**Review by AaronVanGeffen — COMMENTED** [review]
These look like good changes to me, but CI is currently failing. Could you address that?

File: src/OpenLoco/Message.h (hunk 2 above)
Comment (duncanspumpkin) [reply src/OpenLoco/Message.h:62]: Hmm not really its used to indicate if it has any argument so could be entity, station, location, town, ...

File: src/OpenLoco/Message.h (hunk 1 above)
Comment (duncanspumpkin) [reply src/OpenLoco/Message.h]: see later

File: src/OpenLoco/Message.cpp [hunk 3]
```diff
@@ -0,0 +1,165 @@
+#include "Message.h"
+#include "Audio/Audio.h"
+
+namespace OpenLoco
+{
+    // 0x004F8BE4
+    static constexpr uint16_t _MessageTypeFlagsArr[31] = {
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Message.cpp:7]: Could we introduce a constant for this `31`? I've seen it pop up in several places.

File: src/OpenLoco/Message.cpp [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        MessageCriticality::majorCompetitor,
+    };
+
+    MessageCriticality getMessageCriticality(const MessageType type)
+    {
+        return messageCriticalities[enumValue(type)];
+    }
+
+    // 0x004F8C41
+    constexpr static Audio::SoundId messageSounds[31] = {
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Message.cpp:127]: I think we should consider introducing a struct that merges these four tables (messageCriticalities, messageSounds, _MessageTypeFlagsArr, _MessageItemArgumentType).

File: src/OpenLoco/Message.cpp (hunk 3 above)
Comment (duncanspumpkin) [reply src/OpenLoco/Message.cpp:7]: when we know all of the message types we can have one added.

File: src/OpenLoco/Message.cpp (hunk 4 above)
Comment (duncanspumpkin) [reply src/OpenLoco/Message.cpp:127]: already ahead of you :P

File: src/OpenLoco/Message.cpp (hunk 3 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/Message.cpp:48]: Oh, I was under the impression we had already uncovered all types, given the four lookup tables.

**Review by AaronVanGeffen — APPROVED** [review]
Code changes look good. Passes replay tests.

**Pull request merged**